server/models/visit.js

```diff
@@ -0,0 +1,157 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+var q = require('q');
+var uuid = require('uuid');
+
+var postgres = require('../db/postgres');
+var elasticsearch = require('../db/elasticsearch');
+var log = require('../logger')('server.models.visit');
+
+var _verbose = function() {
+  var logline = [].join.call(arguments, ', ');
+  log.verbose(logline);
+};
+
+var visit = {
+  _onFulfilled: function _onFulfilled(msg, callback, results) {
+    _verbose(msg);
+    callback(null, results);
+  },
+  _onRejected: function _onRejected(msg, callback, err) {
+    log.warn(msg);
+    callback(err);
+  },
+  _transform: function _transform(results) {
+    // do a little transformin' and this might barf, so do it before `done`
+
+    // if nothing was found, continue
+    if (!results) { return; }
```

Comment:
ESLint is complaining about "Expected no return value" on line 45. I think it's because up here on line 32 we're just calling `return` with no args or return value, whereas on line 45 we're returning an Object. Not sure if we should return an empty object here, or just ignore whatever ESLint think's it's validating on the [`consistent-return`](http://eslint.org/docs/rules/consistent-return.html) rule.

> _One of the confusing aspects of JavaScript is that any function may or may not return a value at any point in time. When a function exits without any `return` statement executing, the function returns `undefined`. Similarly, calling `return` without specifying any value will cause the function to return `undefined`. Only when `return` is called with a value is there a change in the function's return value. ..._

All moot if your calling code checks for a valid return value vs `undefined` though. Feel free to stop reading my nonsense at any time...